library/Icingadb/Model/Behavior/ReRoute.php

```diff
@@ -39,6 +46,14 @@ public function rewriteCondition(Filter\Condition $condition, $relation = null)
                 );
             }
 
+            if (in_array(substr($relation, 0, -1), self::SPECIAL_RELATIONS) && $remainingPath === 'servicegroup.name') {
+                $applyAll = Filter::all();
+                $applyAll->add(Filter::like($relation . 'object_type', 'host'));
+                $applyAll->add(Filter::like('host.' . $path, $condition->getValue()));
```

Comment:
You can't use `Filter::like` here. The condition might be of any type. It must not change! Use the `$class` variable from above here as well.

Though, maybe just clone `$filter` and change the column name. This way the meta data is preserved and you don't have to transfer it again.

Reply:
Also, prepend `$relation` to `host.`. It must be a fully qualified path that's used by the condition.